geomstats/visualization.py

```diff
@@ -542,3 +559,89 @@ def plot(points, ax=None, space=None,
         klein_disk.draw(ax, **point_draw_kwargs)
 
     return ax
+
+
+class DataArrow():
+    """Arrow class for pointing from some data coordinates to others."""
+
+    def __init__(self, fig):
+        self.fig = fig
+
+    def draw(self, x_from, y_from, x_to, y_to):
+        """
+        Draw the arrow in data coordinates.
+
+        :param x_from: x coordinate at arrow base
+        :param y_from: y coordinate at arrow base
+        :param x_to: x coordinate at arrow head
+        :param y_to: y coordinate at arrow head
+        """
+        fig_trans = self.fig.transFigure.inverted()
+        coord_from = fig_trans.transform(
+            self.fig.axes[0].transData.transform(
+                (x_from, y_from)))
+        coord_to = fig_trans.transform(
+            self.fig.axes[0].transData.transform(
+                (x_to, y_to)))
+        arrow = matplotlib.patches.FancyArrowPatch(
+            coord_from,
+            coord_to,
+            transform=self.fig.transFigure,
+            fc="k",
+            connectionstyle="arc3,rad=0.",
+            arrowstyle='fancy',
+            alpha=1.,
+            mutation_scale=5.)
+        self.fig.patches.append(arrow)
+
+
+class Ellipsis2D():
+    """Plotting class for ellipses representing SPD matrices in 2D."""
+
+    def __init__(self):
+        self.fig = plt.figure()
+        spec = gridspec.GridSpec(ncols=1, nrows=1, figure=self.fig)
+        self.fig.add_subplot(spec[0, 0])
+        self.colors = ['r', 'g', 'b', 'xkcd:camel', 'k']
+        self.colors_alt = ['xkcd:burgundy', 'olive', 'cyan', 'xkcd:mud brown']
+        nice_matplotlib()
+
+    def draw(self, data_point, **kwargs):
+        """Draw the ellipsis."""
+        X, Y, x, y = self.compute_coordinates(data_point)
+        self.fig.axes[0].plot(X, Y, **kwargs)
+        return x, y
+
+    @staticmethod
+    def compute_coordinates(P):
```

Comment:
Avoid single-letter variables when possible. Variable names are also in "lower_cases" style as per pep8.

Reply:
Renamed to `point`.